Failed to write through Logstash, file too large?

One of the log files was 2gb large and no more logs were imported to elasticsearch.
Perhaps "offset" field should be long instead of int.

Here are the log details from Elasticsearch:

```
2015-10-27 00:33:51,905][DEBUG][action.bulk              ] [Collector] [filebeat-2015.10.26][4] failed to execute bulk item (index) index {[filebeat-2015.10.26][log][AVCmgDaGzUbSPeSYXpfB], source[{"message":"{\"meta\":{\"code\":200},\"url\":\"https:\\/\\/api.instagram.com\\/v1\\/media\\/1104599124806528294_1420079922?access_token=2122121522.ae5316e.dc29ec3a6f8a41df955a68f9ef10d547\",\"token\":\"2122121522.ae5316e.dc29ec3a6f8a41df955a68f9ef10d547\",\"ip\":\"94.203.8.122\",\"action\":\"media\",\"reason\":\"action_like\"}","@version":"1","@timestamp":"2015-10-26T23:33:51.482Z","count":1,"fields":null,"fileinfo":{},"line":5260918,"offset":2332147062,"shipper":"colibri-worker","source":"/var/log/colibri/instagram.log","timestamp":"2015-10-26T23:33:51.647Z","type":"log","meta":{"code":200},"url":"https://api.instagram.com/v1/media/1104599124806528294_1420079922?access_token=2122121522.ae5316e.dc29ec3a6f8a41df955a68f9ef10d547","token":"2122121522.ae5316e.dc29ec3a6f8a41df955a68f9ef10d547","ip":"94.203.8.122","action":"media","reason":"action_like"}]}
org.elasticsearch.index.mapper.MapperParsingException: failed to parse [offset]
        at org.elasticsearch.index.mapper.core.AbstractFieldMapper.parse(AbstractFieldMapper.java:411)
        at org.elasticsearch.index.mapper.object.ObjectMapper.serializeValue(ObjectMapper.java:706)
        at org.elasticsearch.index.mapper.object.ObjectMapper.parse(ObjectMapper.java:497)
        at org.elasticsearch.index.mapper.DocumentMapper.parse(DocumentMapper.java:544)
        at org.elasticsearch.index.mapper.DocumentMapper.parse(DocumentMapper.java:493)
        at org.elasticsearch.index.shard.IndexShard.prepareCreate(IndexShard.java:466)
        at org.elasticsearch.action.bulk.TransportShardBulkAction.shardIndexOperation(TransportShardBulkAction.java:418)
        at org.elasticsearch.action.bulk.TransportShardBulkAction.shardOperationOnPrimary(TransportShardBulkAction.java:148)
        at org.elasticsearch.action.support.replication.TransportShardReplicationOperationAction$PrimaryPhase.performOnPrimary(TransportShardReplicationOperationAction.java:574)
        at org.elasticsearch.action.support.replication.TransportShardReplicationOperationAction$PrimaryPhase$1.doRun(TransportShardReplicationOperationAction.java:440)
        at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:36)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
Caused by: org.elasticsearch.common.jackson.core.JsonParseException: Numeric value (2332147062) out of range of int
 at [Source: [B@47313003; line: 1, column: 458]
```

Logstash config file:
```
input {
  beats {
    port => 5044
  }
}

filter {
  json {
    source => "message"
  }
}

output {
  elasticsearch {
    protocol => "http"
    host => "localhost"
    port => "9200"
    sniffing => true
    manage_template => false
    index => "%{[@metadata][index]}"
    document_type => "%{[@metadata][type]}"
  }
}
```

Filebeat config:
```
################### Filebeat Configuration Example #########################

############################# Filebeat ######################################
filebeat:
  # List of prospectors to fetch data.
  prospectors:
    # Each - is a prospector. Below are the prospector specific configurations
    -
      # Paths that should be crawled and fetched. Glob based paths.
      # To fetch all ".log" files from a specific level of subdirectories
      # /var/log/*/*.log can be used.
      # For each file found under this path, a harvester is started.
      # Make sure not file is defined twice as this can lead to unexpected behaviour.
      paths:
        - /var/log/colibri.io/*.log
      # - c:\programdata\elasticsearch\logs\*

      # Configure the file encoding for reading file with international characters
      # supported encodings:
      #   plain, utf-8, utf-16be-bom, utf-16be, utf-16le, big5, gb18030, gbk, hzgb2312,
      #   euckr, eucjp, iso2022jp, shiftjis, iso8859-63, iso8859-6i, iso8859-8e,
      #   iso8859-8i
      #encoding: plain

      # Type of the files. Based on this the way the file is read is decided.
      # The different types cannot be mixed in one prospector
      #
      # Possible options are:
      # * log: Reads every line of the log file (default)
      # * stdin: Reads the standard in
      type: log

      # Optional additional fields. These field can be freely picked
      # to add additional information to the crawled log files for filtering
      #fields:
      #  level: debug
      #  review: 1

      # Ignore files which were modified more then the defined timespan in the past
      # Time strings like 2h (2 hours), 5m (5 minutes) can be used.
      #ignore_older:

      # Scan frequency in seconds.
      # How often these files should be checked for changes. In case it is set
      # to 0s, it is done as often as possible. Default: 10s
      #scan_frequency: 10s

      # Defines the buffer size every harvester uses when fetching the file
      #harvester_buffer_size: 16384

      # Always tail on log rotation. Disabled by default
      # Note: This may skip entries
      #tail_on_rotate: false

    #-
    #  paths:
    #    - /var/log/apache/*.log
    #  type: log
    #
    #  # Ignore files which are older then 24 hours
    #  ignore_older: 24h
    #
    #  # Additional fields which can be freely defined
    #  fields:
    #    type: apache
    #    server: localhost
    #-
    #  type: stdin
    #  paths:
    #    - "-"

  # General filebeat configuration options
  #
  # Event count spool threshold - forces network flush if exceeded
  #spool_size: 1024

  # Defines how often the spooler is flushed. After idle_timeout the spooler is
  # Flush even though spool_size is not reached.
  #idle_timeout: 5s

  # Name of the registry file. Per default it is put in the current working
  # directory. In case the working directory is changed after when running
  # filebeat again, indexing starts from the beginning again.
  registry_file: /var/lib/filebeat/registry

  # Full Path to directory with additional prospector configuration files. Each file must end with .yml
  # These config files must have the full filebeat config part inside, but only
  # the prospector part is processed. All global options like spool_size are ignored.
  # The config_dir MUST point to a different directory then where the main filebeat config file is in.
  config_dir:


############################# Output ##########################################

# Configure what outputs to use when sending the data collected by filebeat.
# You can enable one or multiple outputs by setting enabled option to true.
output:

  ### Elasticsearch as output
  #elasticsearch:

    # Set to true to enable elasticsearch output
    #enabled: false

    # Array of hosts to connect to.
    # Scheme and port can be left out and will be set to the default (http and 9200)
    # In case you specify and additional path, the scheme is required: http://localhost:9200/path
    # IPv6 addresses should always be defined as: https://[2001:db8::1]:9200
    #hosts: []

    # Optional protocol and basic auth credentials. These are deprectad
    #protocol: "https"
    #username: "admin"
    #password: "s3cr3t"

    # Optional index name. The default is "filebeat" and generates
    # [filebeat-]YYYY.MM.DD keys.
    #index: "filebeat"

    # Optional HTTP Path
    #path: "/elasticsearch"

    # The number of times a particular Elasticsearch index operation is attempted. If
    # the indexing operation doesn't succeed after this many retries, the events are
    # dropped. The default is 3.
    #max_retries: 3

    # The maximum number of events to bulk in a single Elasticsearch bulk API index request.
    # The default is 10000.
    #bulk_max_size: 10000

    # The number of seconds to wait for new events between two bulk API index requests.
    # If `bulk_max_size` is reached before this interval expires, addition bulk index
    # requests are made.
    #flush_interval

    # tls configuration
    #tls:
      # If disabled is set to true, the tls section will be ignored and the
      # host its certificate authorities will be used.
      #disabled: true

      # List of root certificates for HTTPS server verifications
      #certificate_authorities: ["/etc/pki/root/ca.pem"]

      # Certificate for TLS client authentication
      #certificate: "/etc/pki/client/cert.pem"

      # Client Certificate Key
      #certificate_key: "/etc/pki/client/cert.key"

      # Controls whether the client verifies server certificates and host name.
      # If insecure is set to true, all server host names and certificates will be
      # accepted. In this mode TLS based connections are susceptible to
      # man-in-the-middle attacks. Use only for testing.
      #insecure: true

      # Configure cipher suites to be used for TLS connections
      #cipher_suites: []

      # Configure curve types for ECDHE based cipher suites
      #curve_types: []


  ### Logstash as output
  logstash:
    # Uncomment out this option if you want to output to Logstash. The default is false.
    enabled: true

    # The Logstash hosts
    hosts: ["somelogstashhost:5044"]

    # Optional load balance the events between the Logstash hosts
    #loadbalance: true

    # Optional index name. The default index name is filebeat.
    #index: filebeat

    # Optional TLS. The default is on
    #tls:
      #disabled: true

      # List of root certificates for HTTPS server verifications
      #certificate_authorities: ["/etc/pki/root/ca.pem"]

      # Certificate for TLS client authentication
      #certificate: "/etc/pki/client/cert.pem"

      # Client Certificate Key
      #certificate_key: "/etc/pki/client/cert.key"

      # Controls whether the client verifies server certificates and host name.
      # If insecure is set to true, all server host names and certificates will be
      # accepted. In this mode TLS based connections are susceptible to
      # man-in-the-middle attacks. Use only for testing.
      #insecure: true

      # Configure cipher suites to be used for TLS connections
      #cipher_suites: []

      # Configure curve types for ECDHE based cipher suites
      #curve_types: []


  ### File as output
  #file:
    # Enabling file output
    #enabled: false

    # Path to the directory where to save the generated files. The option is mandatory.
    #path: "/tmp/filebeat"

    # Name of the generated files. The default is `filebeat` and it generates
    # files: `filebeat`, `filebeat.1`, `filebeat.2`, etc.
    #filename: filebeat

    # Maximum size in kilobytes of each file. When this size is reached, the files are
    # rotated. The default value is 10 MB.
    #rotate_every_kb: 10000

    # Maximum number of files under path. When this number of files is reached, the
    # oldest file is deleted and the rest are shifted from last to first. The default
    # is 7 files.
    #number_of_files: 7


############################# Shipper #########################################

shipper:
  # The name of the shipper that publishes the network data. It can be used to group
  # all the transactions sent by a single shipper in the web interface.
  # If this options is not defined, the hostname is used.
  #name:

  # The tags of the shipper are included in their own field with each
  # transaction published. Tags make it easy to group servers by different
  # logical properties.
  #tags: ["service-X", "web-tier"]

  # Uncomment the following if you want to ignore transactions created
  # by the server on which the shipper is installed. This option is useful
  # to remove duplicates if shippers are installed on multiple servers.
  #ignore_outgoing: true


############################# Logging #########################################

#logging:
#  selectors: []
#
#  # Rotator config
#  files:
#    path:
#    name:
#    rotateEveryBytes:
#    keepFiles:
#  to_syslog: false
#  to_files: false
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -12,6 +12,7 @@ All documentation about Filebeat can be found here.
 
 ### Bugfixes
 - Omit 'fields' from event JSON when null. #126
+- Make offset and line value of type long in elasticsearch template to prevent overflow #140
 
 ### Added
 

--- a/etc/filebeat.template.json
+++ b/etc/filebeat.template.json
@@ -29,11 +29,11 @@
           "index": "analyzed"
         },
         "line": {
-          "type": "integer",
+          "type": "long",
           "doc_values": "true"
         },
         "offset": {
-          "type": "integer",
+          "type": "long",
           "doc_values": "true"
         }
       }

